File: agent/HardBans.cs
namespace PrivGate.Agent;

public static class HardBans
{
    public static readonly HashSet<string> Names = new(StringComparer.OrdinalIgnoreCase)
    {
        "cmd.exe", "powershell.exe", "pwsh.exe", "wscript.exe", "mshta.exe", "reg.exe"
    };

    public static bool IsBanned(string filePath)
    {
        var name = Path.GetFileName(filePath);
        return Names.Contains(name);
    }
}

File: src/lib/hard-bans.ts
export const HARD_BANS = [
  "cmd.exe",
  "powershell.exe",
  "pwsh.exe",
  "wscript.exe",
  "mshta.exe",
  "reg.exe",
] as const;

export function fileNameOf(filePath: string): string {
  const normalized = filePath.replaceAll("\\", "/");
  const base = normalized.split("/").pop() ?? filePath;
  return base.toLowerCase();
}

export function isHardBanned(filePath: string): boolean {
  return (HARD_BANS as readonly string[]).includes(fileNameOf(filePath));
}
